storybook/src/components/modal/modal.core-template.ts

```diff
@@ -1,49 +1,36 @@
-import { DsoModalController } from "@dso-toolkit/core";
 import { Modal } from "dso-toolkit";
-import { html, TemplateResult, render } from "lit-html";
+import { html, TemplateResult } from "lit-html";
+
+import { ifDefined } from "lit-html/directives/if-defined.js";
 
 import { ComponentImplementation } from "../../templates";
 
 export const coreModal: ComponentImplementation<Modal<TemplateResult>> = {
   component: "modal",
   implementation: "core",
   template: () =>
-    function modalTemplate({ modalTitle, role, showCloseButton, initialFocus, body, footer, dsoClose, returnFocus }) {
-      const open = () => {
-        /* eslint-disable @typescript-eslint/no-explicit-any */
-        if ((window as any).modalRef) {
-          (window as any).modalRef.close();
-          delete (window as any).modalRef;
-        }
-        /* eslint-enable @typescript-eslint/no-explicit-any */
-
-        const controller = new DsoModalController();
-
-        const bodyElement = document.createDocumentFragment();
-        const footerElement = document.createDocumentFragment();
-
-        render(body, bodyElement);
-        render(footer, footerElement);
-
-        const modalRef = controller.open(
-          {
-            title: modalTitle,
-            body: bodyElement,
-            footer: footerElement.childElementCount ? footerElement : undefined,
-          },
-          { role, showCloseButton, initialFocus, returnFocus }
-        );
-
-        if (dsoClose) {
-          modalRef.addEventListener("dsoClose", dsoClose as EventListenerOrEventListenerObject);
-        }
-
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (window as any)["modalRef"] = modalRef;
-      };
-
-      open();
-
-      return html``;
+    function modalTemplate({
+      fullscreen,
+      modalTitle,
+      role,
+      showCloseButton,
+      initialFocus,
+      body,
+      footer,
+      dsoClose,
+    }) {
+      return html`
+        <dso-modal
+          role=${role}
```

Comment:
Ik weet dat we dit al deden, maar is dit wel handig? `role` op `<dso-modal>` ? Ik bedoel, dit doet dan toch ook "iets"? @hansgrimm ?

Reply:
Goed punt, maar voor nu even handhaven, want nog niet alle AT’s snappen `<dialog>`. Zo vang je dat op.

Reply:
Prima